src/java.base/share/classes/jdk/internal/misc/JitCompiler.java

```diff
@@ -0,0 +1,123 @@
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.misc;
+
+import jdk.internal.vm.annotation.IntrinsicCandidate;
+
+/**
+ * Just-in-time-compiler-related queries
+ */
+public class JitCompiler {
+    /**
+     * Determine if {@code expr} can be evaluated to a constant value by the JIT
+     * compiler. For example, {@code isCompileConstant(5)} will be evaluated
+     * to {@code true} by the JIT compiler, while
+     * {@code isCompileConstant(random.nextLong())} will likely be evaluated
+     * to {@code false}.
+     *
+     * <p>Note that the JIT compiler is responsible to change the return value
+     * of this method to {@code true}, the interpreter always returns {@code false}.
+     *
+     * <p>Given the nondeterministic nature of this method, the result of the
+     * program must not depend on the return value of this method. It must be
+     * used as a pure optimization to take advantage of the constant nature of
+     * {@code expr}. E.g. for a runtime variable, looking up a hashmap may be
+     * the most efficient look up method, however, if the look up table is
+     * constant, it may be better to use a chain of if-else in cases where the
+     * input is also a constant.
+     *
+     * @param expr the expression to be evaluated
+     * @return {@code true} if the JIT compiler determines the {@code expr} is
+     *         always evaluated to a constant value, {@code false} otherwise
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(boolean expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(byte expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(short expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(char expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(int expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(long expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(float expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
+    public static boolean isCompileConstant(double expr) {
+        return false;
+    }
+
+    /**
+     * @see #isCompileConstant(boolean)
+     */
+    @IntrinsicCandidate
```

Comment:
Note how the Java entry for MH intrinsic we have replaced had `@Hidden`. These methods should have `@Hidden` too then? Probably applies to other entries too.

Reply:
I don't understand why this needs to be `@Hidden`, the javadoc says that a function annotated with `@Hidden` is omitted from the stacktraces. This function does not call anything so what is the point of hiding it?

Reply:
I suspect there is a code that counts stack traces somewhere that relies on it in MH parts. There is no harm for doing `@Hidden` here, I think.